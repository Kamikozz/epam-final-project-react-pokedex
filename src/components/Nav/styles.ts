import { createStyles } from "@material-ui/core";

const styles = createStyles({
  root: {
    flexGrow: 1
  },
  tabsIndicator: {
    backgroundColor: "red"
  },
  tabsRoot: {
    color: "red"
  }
});

export default styles;
