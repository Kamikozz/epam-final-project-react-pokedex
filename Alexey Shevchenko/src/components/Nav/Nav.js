import React from 'react';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import {withStyles} from '@material-ui/core/styles';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
// import PhoneIcon from '@material-ui/icons/Phone';
// import PersonPinIcon from '@material-ui/icons/PersonPin';
import pokedexIcon from '../../assets/logo.png'
import './Nav.css';
import {Link} from "react-router-dom";

const styles = {
	root: {
		flexGrow: 1
	},
	tabsIndicator: {
		backgroundColor: 'red'
	},
	tabsRoot: {
		color: 'red'
	},
};

class Nav extends React.Component {
	state = {
		value: 0,
		currentUserId: 1,
	};

	handleChange = (event, value) => {
		this.setState({value});
	};

	render() {
		const {classes} = this.props;

		return (
			<Paper square className={classes.root}>
				<Tabs
					classes={{root: classes.tabsRoot, indicator: classes.tabsIndicator }}
					value={this.state.value}
					onChange={this.handleChange}
					variant="fullWidth"
				>
					<Tab
						icon={
							<img src={pokedexIcon} alt="Home pokedex" width="24px"/>
						}
						label="POKEDEX" component={Link} to="/" />
					<Tab label="GOTCHA!" component={Link} to={`/caught_pokemons`}  />
				</Tabs>
			</Paper>
		);
	}
}

Nav.propTypes = {
	classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Nav);