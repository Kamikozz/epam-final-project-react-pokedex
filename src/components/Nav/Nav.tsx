import React, { useState } from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import { WithStyles } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import Paper from "@material-ui/core/Paper";
import Tabs from "@material-ui/core/Tabs";
import Tab from "@material-ui/core/Tab";

import pokedexIcon from "../../assets/pokeball.png";
import caughtPokemonsIcon from "../../assets/pokeball3.png";
import routes from "../../routes";
import styles from "./styles";

interface Props extends WithStyles<typeof styles> {
  selected: number;
  classes: {
    root: string;
    tabsRoot: string;
    tabsIndicator: string;
  };
};

const LinkToPokemonsPage = (props: any) => <Link to={routes.pokemonsPage} {...props} />;
const LinkToCaughtPokemonsPage = (props: any) => <Link to={routes.caughtPokemonsPage} {...props} />;

const Nav = (props: Props) => {
  const [selectedTabIndex, setSelectedTabIndex] = useState(props.selected);
  const handleChange = (event: any, selectedTabIndex: number) => {
    setSelectedTabIndex(selectedTabIndex);
  };
  const { classes } = props;
  return (
    <Paper square className={classes.root}>
      <Tabs
        classes={{ root: classes.tabsRoot, indicator: classes.tabsIndicator }}
        value={selectedTabIndex}
        onChange={handleChange}
        variant="fullWidth"
      >
        <Tab
          icon={<img src={pokedexIcon} alt="Home pokedex" width="24px" />}
          label="POKEDEX"
          component={LinkToPokemonsPage}
        />
        <Tab
          icon={
            <img
              src={caughtPokemonsIcon}
              alt="Caught pokemons"
              width="24px"
            />
          }
          label="GOTCHA!"
          component={LinkToCaughtPokemonsPage}
        />
      </Tabs>
    </Paper>
  );
};

Nav.propTypes = {
  classes: PropTypes.object.isRequired
} as any;

export default withStyles(styles)(Nav);
