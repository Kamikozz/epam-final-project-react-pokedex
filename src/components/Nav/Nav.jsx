import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import { withStyles } from "@material-ui/core/styles";
import Paper from "@material-ui/core/Paper";
import Tabs from "@material-ui/core/Tabs";
import Tab from "@material-ui/core/Tab";

import pokedexIcon from "../../assets/pokeball.png";
import "./Nav.css";

const styles = {
  root: {
    flexGrow: 1
  },
  tabsIndicator: {
    backgroundColor: "red"
  },
  tabsRoot: {
    color: "red"
  }
};

class Nav extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event, selectedTabIndex) {
    this.setState({ selectedTabIndex });
  }

  render() {
    const { classes } = this.props;

    return (
      <Paper square className={classes.root}>
        <Tabs
          classes={{ root: classes.tabsRoot, indicator: classes.tabsIndicator }}
          value={this.props.selected}
          onChange={this.handleChange}
          variant="fullWidth"
        >
          <Tab
            icon={<img src={pokedexIcon} alt="Home pokedex" width="24px" />}
            label="POKEDEX"
            component={Link}
            to="/"
          />
          <Tab label="GOTCHA!" component={Link} to={"/caught_pokemons"} />
        </Tabs>
      </Paper>
    );
  }
}

Nav.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(Nav);
