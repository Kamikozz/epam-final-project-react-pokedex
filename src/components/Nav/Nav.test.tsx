import React from 'react';
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import { render } from '@testing-library/react';

import Nav from './Nav';

it('renders without crashing', () => {
  render(
    <Router>
      <Nav selected={0} />
    </Router>
  );
});

describe('matches snapshot', () => {
  it('with selected=0', () => {
    const { container } = render(
      <Router>
        <Nav selected={0} />
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('with selected=1', () => {
    const { container } = render(
      <Router>
        <Nav selected={1} />
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('with selected more than number of tabs', () => {
    const { container } = render(
      <Router>
        <Nav selected={10} />
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

