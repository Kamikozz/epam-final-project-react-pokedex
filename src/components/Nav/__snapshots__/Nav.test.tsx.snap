// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with selected more than number of tabs 1`] = `
<div
  class="MuiPaper-root-196 MuiPaper-elevation2-200 Nav-root-193"
>
  <div
    class="MuiTabs-root-223 Nav-tabsRoot-195"
  >
    <div
      class="MuiTabs-flexContainer-224"
    >
      <div
        class="MuiTabs-scroller-226 MuiTabs-fixed-227"
        role="tablist"
        style="margin-bottom: 0px;"
      >
        <div
          class="MuiTabs-flexContainer-224"
        >
          <a
            aria-selected="false"
            class="MuiButtonBase-root-244 MuiTab-root-232 MuiTab-textColorInherit-234 MuiTab-labelIcon-233 MuiTab-fullWidth-239"
            href="/"
            role="tab"
            tabindex="0"
          >
            <span
              class="MuiTab-wrapper-240"
            >
              <img
                alt="Home pokedex"
                src="pokeball.png"
                width="24px"
              />
              <span
                class="MuiTab-labelContainer-241"
              >
                <span
                  class="MuiTab-label-242"
                >
                  POKEDEX
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root-247"
            />
          </a>
          <a
            aria-selected="false"
            class="MuiButtonBase-root-244 MuiTab-root-232 MuiTab-textColorInherit-234 MuiTab-labelIcon-233 MuiTab-fullWidth-239"
            href="/caught_pokemons/"
            role="tab"
            tabindex="0"
          >
            <span
              class="MuiTab-wrapper-240"
            >
              <img
                alt="Caught pokemons"
                src="pokeball3.png"
                width="24px"
              />
              <span
                class="MuiTab-labelContainer-241"
              >
                <span
                  class="MuiTab-label-242"
                >
                  GOTCHA!
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root-247"
            />
          </a>
        </div>
        <span
          class="MuiPrivateTabIndicator-root-254 MuiPrivateTabIndicator-colorSecondary-256 MuiTabs-indicator-231 Nav-tabsIndicator-194"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`matches snapshot with selected=0 1`] = `
<div
  class="MuiPaper-root-68 MuiPaper-elevation2-72 Nav-root-65"
>
  <div
    class="MuiTabs-root-95 Nav-tabsRoot-67"
  >
    <div
      class="MuiTabs-flexContainer-96"
    >
      <div
        class="MuiTabs-scroller-98 MuiTabs-fixed-99"
        role="tablist"
        style="margin-bottom: 0px;"
      >
        <div
          class="MuiTabs-flexContainer-96"
        >
          <a
            aria-selected="true"
            class="MuiButtonBase-root-116 MuiTab-root-104 MuiTab-textColorInherit-106 MuiTab-selected-109 MuiTab-labelIcon-105 MuiTab-fullWidth-111"
            href="/"
            role="tab"
            tabindex="0"
          >
            <span
              class="MuiTab-wrapper-112"
            >
              <img
                alt="Home pokedex"
                src="pokeball.png"
                width="24px"
              />
              <span
                class="MuiTab-labelContainer-113"
              >
                <span
                  class="MuiTab-label-114"
                >
                  POKEDEX
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root-122"
            />
          </a>
          <a
            aria-selected="false"
            class="MuiButtonBase-root-116 MuiTab-root-104 MuiTab-textColorInherit-106 MuiTab-labelIcon-105 MuiTab-fullWidth-111"
            href="/caught_pokemons/"
            role="tab"
            tabindex="0"
          >
            <span
              class="MuiTab-wrapper-112"
            >
              <img
                alt="Caught pokemons"
                src="pokeball3.png"
                width="24px"
              />
              <span
                class="MuiTab-labelContainer-113"
              >
                <span
                  class="MuiTab-label-114"
                >
                  GOTCHA!
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root-122"
            />
          </a>
        </div>
        <span
          class="MuiPrivateTabIndicator-root-119 MuiPrivateTabIndicator-colorSecondary-121 MuiTabs-indicator-103 Nav-tabsIndicator-66"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`matches snapshot with selected=1 1`] = `
<div
  class="MuiPaper-root-132 MuiPaper-elevation2-136 Nav-root-129"
>
  <div
    class="MuiTabs-root-159 Nav-tabsRoot-131"
  >
    <div
      class="MuiTabs-flexContainer-160"
    >
      <div
        class="MuiTabs-scroller-162 MuiTabs-fixed-163"
        role="tablist"
        style="margin-bottom: 0px;"
      >
        <div
          class="MuiTabs-flexContainer-160"
        >
          <a
            aria-selected="false"
            class="MuiButtonBase-root-180 MuiTab-root-168 MuiTab-textColorInherit-170 MuiTab-labelIcon-169 MuiTab-fullWidth-175"
            href="/"
            role="tab"
            tabindex="0"
          >
            <span
              class="MuiTab-wrapper-176"
            >
              <img
                alt="Home pokedex"
                src="pokeball.png"
                width="24px"
              />
              <span
                class="MuiTab-labelContainer-177"
              >
                <span
                  class="MuiTab-label-178"
                >
                  POKEDEX
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root-186"
            />
          </a>
          <a
            aria-selected="true"
            class="MuiButtonBase-root-180 MuiTab-root-168 MuiTab-textColorInherit-170 MuiTab-selected-173 MuiTab-labelIcon-169 MuiTab-fullWidth-175"
            href="/caught_pokemons/"
            role="tab"
            tabindex="0"
          >
            <span
              class="MuiTab-wrapper-176"
            >
              <img
                alt="Caught pokemons"
                src="pokeball3.png"
                width="24px"
              />
              <span
                class="MuiTab-labelContainer-177"
              >
                <span
                  class="MuiTab-label-178"
                >
                  GOTCHA!
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root-186"
            />
          </a>
        </div>
        <span
          class="MuiPrivateTabIndicator-root-183 MuiPrivateTabIndicator-colorSecondary-185 MuiTabs-indicator-167 Nav-tabsIndicator-130"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
  </div>
</div>
`;
